README.md

```diff
@@ -44,10 +47,61 @@ pip3 uninstall kompos --yes
 pip3 install --upgrade kompos
 ```
 
+### Nix integration
 
-### Nix versioning
+With kompos you can leverage [nix](https://nixos.org/nix/) to pin your
+infrastructure code (i.e terraform & helmfile releases) on a specific version.
+This enables you to finely control your deployments and use different
+infrastructure versions per environment, cluster etc.
+
+Enable the integration with the `--nix` flag with each command. Below are the
+necessary parts of komposconfig regarding nix/versioning:
+
+```yaml
+terraform:
+  # This is the place to look for the terraform repo locally.
+  # Used as the default if nix is not enabled. 
+  local_path: '/home/user/terraform-stack'
+
+  # This is needed in case the modules are not in the root of the repo.
+  root_path: 'src/terraform'
+
+  repo:
+    # This will be the name of the nix derivation for terraform.
+    name: 'terraform-stack'
+
+    # The repo we would like to version.
+    url:  "git@github.com:my-org/terraform-stack.git"
+
+# Likewise for helmfile.
+helmfile:
+  local_path: '/home/user/helmfile-releases'
+  root_path: 'src/helmfiles'
+
+  repo:
+    name: 'helmfile-releases'
+    url:  "git@github.com:my-org/helmfile-releases.git"
 ```
-curl https://nixos.org/nix/install | sh
+
+And in the hierarchical configuration you'll need the following key/value pairs:
+
+```yaml
+infrastructure:
+  terraform:
+    version: "0.1.0" # A git tag or a commit sha.
+
+    # Optional. 
+    # The sha256 hash of the repo needed by nix's fetchgit function.
+    #
+    # It can be obtained using nix-prefetch-git:
+    # $ nix-prefetch-git --url <repo-url> --rev <version>
+    # 
+    # If ommitted nix-prefetch-git will be used to retrieve it.
+    sha256: "ab9190b0ecc8060a54f1fac7de606e6b2c4757c227f2ce529668eb145d9a9516"
```

Comment:
not sure I understand when should I (the user) provide both `version` and `sha256`?

Reply:
`sha256` is there to provide data integrity, so you'll always have the same input. One use-case where this can be omitted is when you have a tag that get's updated periodically e.g `my-project-nightly`.

Reply:
makes sense, add this info as a comment